### ISSUE TITLE
RDKit descriptors

### ISSUE THREAD

**Issue by femisegvn** [femisegvn/antiviral-potency#2]
Problem with trying to apply all 217 descriptors -- one of the values is stupidly large (like 10^48) so leads to ridiculous results. I should attempt to identify and resolve this issue, before actually testing with all of the descriptors because i think it was slightly unfair to the MLP to only use like 8 descriptors.

Furthermore, I think the RDKit descriptors should be scaled to fit between 0 and 1 and that might significantly help the MLP.

**Comment by femisegvn** [comment]
preprocessing with `sklearn.Preprocessing.MaxAbsScaler()` significantly improved MLP performance -- turns out scaling isn't just a waste of time ... duh.

**Comment by femisegvn** [comment]
Now i need to:
1. implement the scaling properly -- fit on the training set, transform on the test set
2. implement a feature selection method -- e.g. determining some correlation between the list of rdkit features and the endpoints and selecting the top n features
3. see if feature selection outperforms throwing all of the features in at once
4. Benchmark again against the other featurisation schemes.

**Comment by femisegvn** [comment]
scaling has been implemented -- literally only 3 lines.., but still throwing all the descriptors results in ridiculous performance. I probably ought to find out what descriptor is causing this nonsense.

**Comment by femisegvn** [comment]
2. implemented f-test for feature selection
3. Feature selection does indeed outperform all features because of those massive outliers (which I still ought to identify)
4. feature selection improved MLP performance, but after hypopt and testing, XGBoost is still better
